exonum/src/runtime/rust/interfaces.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Important interservice communication interfaces.
+
+use crate::{
+    merkledb::{BinaryValue, Snapshot},
+    runtime::{CallContext, Caller, DispatcherError, MethodId, SUPERVISOR_SERVICE_ID},
+};
+
+use super::{ExecutionError, Interface, TransactionContext};
+
+/// Fully qualified name of the ['Configure`] interface.
+///
+/// ['Configure`]: trait.Configure.html
+pub const CONFIGURE_INTERFACE_NAME: &str = "Configure";
+/// Identifier of the [`Configure::verify_config`] method.
+///
+/// [`Configure::verify_config`]: trait.Configure.html#tymethod.verify_config
+pub const VERIFY_CONFIG_METHOD_ID: MethodId = 0;
+/// Identifier of the [`Configure::apply_config`] method.
+///
+/// [`Configure::apply_config`]: trait.Configure.html#tymethod.apply_config
+pub const APPLY_CONFIG_METHOD_ID: MethodId = 1;
+
+/// Describes a procedure for updating the configuration of a service instance.
+pub trait Configure {
+    /// The specific type of parameters passed during the service instance configuration.    
+    type Params: BinaryValue;
+    /// Verify a new configuration parameters before before their actual application.
+    ///
+    /// This method is called by the new configuration change proposal. If the proposed
+    /// parameters do not fit for this service instance, it should return a corresponding
+    /// error to discard this proposal. Thus only a configuration change proposal in which all
+    /// changes are correct can be applied later.
+    ///
+    /// The proposal approval process details, and even the configuration proposal format, depends
+    /// on the particular implementation.
+    ///
+    /// # Execution policy
+    ///
+    /// At the moment, this method can only be called on behalf of the supervisor service instance.
+    /// In other words, only a method with the specified [identifier] can call this method.
+    ///
+    /// [identifier]: ../../constant.SUPERVISOR_SERVICE_ID.html
+    fn verify_config(
+        &self,
+        context: TransactionContext,
+        params: Self::Params,
+    ) -> Result<(), ExecutionError>;
+    /// Update service configuration with the given parameters.
+    ///
+    /// The configuration parameters passed to the method are discarded immediately.
```

Comment:
Again, `discarded` means "get rid of (someone or something) as no longer useful or desirable". This is obviously not the case. I suggest `The configuration parameters passed to this method are not saved by the dispatcher, hence the service implementation must save them if needed.`

cc @Maria-Nosyk